### PR TITLE
Add scaling integration tests to all charms

### PR DESCRIPTION
**Pull request by ghislainbourgeois**

### CODE REVIEW

File: orchestrator-bundle/orc8r-orchestrator-operator/tests/unit/test_charm.py [hunk 1]
```diff
@@ -193,6 +194,37 @@ def test_given_relations_ready_and_container_can_connect_when_on_pebble_ready_ev
         ]
         self.assertIn(call_command, args)
 
+    @patch("ops.model.Container.exists")
+    @patch("ops.model.Container.exec", new_callable=Mock)
+    @patch("ops.model.Container.push")
+    def test_given_relations_ready_and_container_can_connect_and_unit_is_not_leader_when_on_pebble_ready_event_then_orchestrator_admin_user_is_created(  # noqa: E501
```
Comment (saltiyazan) [inline orchestrator-bundle/orc8r-orchestrator-operator/tests/unit/test_charm.py]: Didn't you mean to say `is_not_created`?
```suggestion
    def test_given_relations_ready_and_container_can_connect_and_unit_is_not_leader_when_on_pebble_ready_event_then_orchestrator_admin_user_is_not_created(  # noqa: E501
```

File: orchestrator-bundle/orc8r-orchestrator-operator/src/charm.py [hunk 2]
```diff
@@ -501,7 +501,8 @@ def _on_magma_orc8r_orchestrator_pebble_ready(
             event.defer()
             return
         self._configure_orc8r(event)
-        self._create_orchestrator_admin_user()
+        if self.unit.is_leader():
+            self._create_orchestrator_admin_user()
```
Comment (saltiyazan) [inline orchestrator-bundle/orc8r-orchestrator-operator/src/charm.py:533]: we have similar behaviour in `nms-magmalte-operator` when calling `_create_master_nms_admin_user`, might need to add the same condition there as well.

File: orchestrator-bundle/orc8r-orchestrator-operator/tests/unit/test_charm.py (hunk 1 above)
Comment (ghislainbourgeois) [reply orchestrator-bundle/orc8r-orchestrator-operator/tests/unit/test_charm.py]: Yes, good catch!

File: orchestrator-bundle/orc8r-orchestrator-operator/src/charm.py (hunk 2 above)
Comment (ghislainbourgeois) [reply orchestrator-bundle/orc8r-orchestrator-operator/src/charm.py:533]: Good point. For some reason, it still passed the integration test for scaling, but I will add the fix, I see that it is also using `StoredState`. Thanks!

**Review by gruyaume — APPROVED** [review]

File: orchestrator-bundle/orc8r-orchestrator-operator/src/charm.py (hunk 2 above)
Comment (ghislainbourgeois) [reply orchestrator-bundle/orc8r-orchestrator-operator/src/charm.py:533]: I have updated `nms-magmalte-operator`, please take a look when you get the chance.

File: orchestrator-bundle/nms-magmalte-operator/tests/unit/test_charm.py [hunk 3]
```diff
@@ -259,11 +263,60 @@ def test_given_relations_not_created_when_juju_action_then_get_admin_credentials
...
+        self.harness.charm._on_get_master_admin_credentials(action_event)
+
+        self.assertEqual(
+            action_event.fail.call_args,
+            [("Peer relation not created yet",)],
+        )
+
+    @patch("ops.model.Container.get_service", new=Mock())
+    @patch("ops.charm.ActionEvent")
+    def test_given_workload_service_is_running_and_unit_is_leader_when_get_admin_credentials_action_then_username_and_password_are_returned(  # noqa: E501
```
Comment (saltiyazan) [inline orchestrator-bundle/nms-magmalte-operator/tests/unit/test_charm.py:310]: I think `test_given_username_email_and_password_are_provided_when_create_nms_admin_user_juju_action_then_pebble_command_is_executed` needs to be updated as well to test if unit is leader.

Comment (ghislainbourgeois) [reply orchestrator-bundle/nms-magmalte-operator/tests/unit/test_charm.py:310]: You are right, updated

**Review by saltiyazan — APPROVED** [review]

File: orchestrator-bundle/nms-magmalte-operator/src/charm.py [hunk 4]
```diff
@@ -470,9 +483,20 @@ def _get_admin_password(self) -> str:
         Returns:
             str: Password
         """
-        if not self._stored.admin_password:
-            self._stored.admin_password = self._generate_password()
-        return self._stored.admin_password
+        app_data = self.model.get_relation("replicas").data[self.app]  # type: ignore[union-attr]
+        if not app_data.get("admin_password"):
+            return ""
```
Comment (gruyaume) [inline orchestrator-bundle/nms-magmalte-operator/src/charm.py:488]: May be cleaner to return None and the return type to be Optional[str]

File: orchestrator-bundle/nms-magmalte-operator/src/charm.py [hunk 5]
```diff
@@ -470,9 +483,20 @@ def _get_admin_password(self) -> str:
         Returns:
             str: Password
         """
-        if not self._stored.admin_password:
-            self._stored.admin_password = self._generate_password()
-        return self._stored.admin_password
+        app_data = self.model.get_relation("replicas").data[self.app]  # type: ignore[union-attr]
```
Comment (gruyaume) [inline orchestrator-bundle/nms-magmalte-operator/src/charm.py:486]: `_get_admin_password` should probably be a property (ex. `_admin_password`)

File: orchestrator-bundle/nms-magmalte-operator/src/charm.py (hunk 4 above)
Comment (ghislainbourgeois) [reply orchestrator-bundle/nms-magmalte-operator/src/charm.py:488]: Opened a new PR for this

File: orchestrator-bundle/nms-magmalte-operator/src/charm.py (hunk 5 above)
Comment (ghislainbourgeois) [reply orchestrator-bundle/nms-magmalte-operator/src/charm.py:486]: Opened a new PR for this

**Pull request merged**